mautrix_signal/web/provisioning_api.py

```diff
@@ -43,17 +43,27 @@ def __init__(self, bridge: "SignalBridge", shared_secret: str) -> None:
         self.bridge = bridge
         self.app = web.Application()
         self.shared_secret = shared_secret
+
+        # Whoami
         self.app.router.add_get("/api/whoami", self.status)
+
+        # Logout
+        self.app.router.add_options("/api/logout", self.login_options)
+        self.app.router.add_post("/api/logout", self.logout)
+
+        # Link API (will be deprecated soon)
         self.app.router.add_options("/api/link", self.login_options)
         self.app.router.add_options("/api/link/wait", self.login_options)
-        # self.app.router.add_options("/api/register", self.login_options)
-        # self.app.router.add_options("/api/register/code", self.login_options)
-        self.app.router.add_options("/api/logout", self.login_options)
         self.app.router.add_post("/api/link", self.link)
         self.app.router.add_post("/api/link/wait", self.link_wait)
-        # self.app.router.add_post("/api/register", self.register)
-        # self.app.router.add_post("/api/register/code", self.register_code)
-        self.app.router.add_post("/api/logout", self.logout)
+
+        # New Login API
+        self.app.router.add_options("/api/link/v2/new", self.login_options)
+        self.app.router.add_options("/api/link/v2/waitForScan", self.login_options)
+        self.app.router.add_options("/api/link/v2/waitForAccount", self.login_options)
+        self.app.router.add_post("/api/link/v2/new", self.link_new)
+        self.app.router.add_post("/api/link/v2/waitForScan", self.link_wait_for_scan)
+        self.app.router.add_post("/api/link/v2/waitForAccount", self.link_wait_for_account)
```

Comment:
`/_matrix/provision/v1/api/link/v2/...` :upside_down_face: 

I'll try to figure out the best way to make the endpoint names less messy

Reply:
Right, https://github.com/mautrix/signal/pull/199/commits/038fc52be38e27f9886ac27709255c61640e33b4 should make versioning sane. Legacy endpoints are unchanged, new endpoints are `/_matrix/provision/v1/link/wait/scan` (and `/new` and `/wait/account`).

There are v2 versions of whoami and logout to remove the redundant `/api` prefix. New versions can also be added easily if ever needed since the version is now in the endpoint path rather than global config file.

Reply:
I think that makes sense. I had actually thought about using `/link/wait/scan` and `/link/wait/account` as well, but I was 50/50 on which to do. I opted for a bit more explicit `v2`, but it makes a lot more sense having the `v2` before `/link`.